src/alchemlyb/tests/parsing/test_gmx.py

```diff
@@ -195,3 +198,60 @@ def test_extract_dHdl_unit():
     dhdl = extract_dHdl(dataset['data']['Coulomb'][0], 310)
     assert dhdl.attrs['temperature'] == 310
     assert dhdl.attrs['energy_unit'] == 'kT'
+
+class TestRobustGMX():
+    '''Test dropping the row that is wrong in different way'''
+    @staticmethod
+    @pytest.fixture(scope='class')
+    def data():
+        dhdl = extract_dHdl(load_benzene()['data']['Coulomb'][0], 310)
+        with bz2.open(load_benzene()['data']['Coulomb'][0], "rt") as bz_file:
+            text = bz_file.read()
+        return text, len(dhdl)
+
+    def test_sanity(self, data, tmpdir):
```

Comment:
Instead of `tmpdir`, it's now [recommended to use `tmp_path`](https://docs.pytest.org/en/6.2.x/tmpdir.html) and then 
```python
new_text = tmp_path / "text.xvg"
```